### PR TITLE
Add samples for retrieval with Identity & Semantic Enforcement

### PR DESCRIPTION
**Pull request by Raj725**
Add samples for retrieval with Identity & Semantic Enforcement

### CODE REVIEW

File: pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            SemanticContext(**semantic_context) if semantic_context else None
+        )
+        # Prepare retriever QA chain
+        retriever = PebbloRetrievalQA.from_chain_type(
+            llm=self.llm,
+            chain_type="stuff",
+            retriever=self.vectordb.as_retriever(),
+            verbose=True,
+            auth_context=auth_context,
+            semantic_context=semantic_context,
```
Comment (shreyas-damle) [inline pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py:108]: Is it possible to hide complexity of preparing semantic_context? It will reduce number of lines RAG app developer has to write to include semantic enforcement. What if we pass `topics_to_deny` and `entities_to_deny` to ` PebbloRetrievalQA.from_chain_type()`?

File: pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
+# Fill-in OPENAI_API_KEY in .env file in this directory before proceeding
```
Comment (shreyas-damle) [inline pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py:1]: Is this file really needed under identity-rag? I think it's primary aim is semantic enforcement, we are adding identity as well within it though, but they are not dependent on each other.

File: pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    )
+
+    while True:
+        print(
+            "Please enter end user details and semantic filters below...\n"
+            "(Leave empty if you don't want to enforce any identity or semantic filters)"
+        )
+        end_user_email_address = input("User email address (Optional): ") or None
+        # topic_to_deny = input("Topics to deny, comma separated (Optional): ") or None
+        # entity_to_deny = input("Entities to deny, comma separated (Optional): ") or None
```
Comment (shreyas-damle) [inline pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py:139]: nit pick - please remove comments.

File: pebblo_cloud/langchain/rag-sample/semantic_rag_sample.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            collection_name=self.qdrant_collection_name,
+        )
+        print(f"Added {len(documents)} documents to Qdrant ...\n")
+        return vectordb
+
+
+if __name__ == "__main__":
+    print("Loading documents to Qdrant ...")
+    # def_folder_id = "15CyFIWOPJOR5BxDID7G6tUisfHU1szrg" # hr-knowledge-base
+    def_folder_id = "1sR3wRgX9hGCdjtD1Vkl-b3hZuzGUP2EU" # Patent Documents
```
Comment (shreyas-damle) [inline pebblo_cloud/langchain/rag-sample/semantic_rag_sample.py:78]: We can remove these values now as we are merging it to main.

**Review by shreyas-damle — COMMENTED** [review]
Few questions:

- What benefit w.r.t semantic enforcement, end user is having here by adding API key? What if we do not include Pebblo API key in this example?
- We need to show connection between `load_semantic = True` in `PebbloSafeLoader` with `semantic_context` either through comments or through readme.

File: pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py (hunk 2 above)
Comment (Raj725) [reply pebblo_saferetriever/langchain/identity-rag/pebblo_semantic_rag.py]: Moved to a new directory

**Review by srics — APPROVED** [review]
